python/src/synapse/ml/dl/DeepVisionClassifier.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (C) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in project root for information.
+
+from pyspark import keyword_only
+from pyspark.ml.param.shared import Param, Params
+from synapse.ml.dl.LightningEstimator import LightningEstimator
+from synapse.ml.dl.LitDeepVisionModel import LitDeepVisionModel
+
+
+class DeepVisionClassifier(LightningEstimator):
+
+    backbone = Param(
+        Params._dummy(), "backbone", "backbone of the deep vision classifier"
+    )
+
+    num_layers_to_fine_tune = Param(
+        Params._dummy(),
+        "num_layers_to_fine_tune",
+        "number of last layers to fine tune for the model, should be smaller or equal to 3",
+    )
+
+    num_classes = Param(Params._dummy(), "num_classes", "number of target classes")
+
+    loss_name = Param(
+        Params._dummy(),
+        "loss_name",
+        "string representation of torch.nn.functional loss function for the underlying pytorch_lightning model, e.g. binary_cross_entropy",
+    )
+
+    optimizer_name = Param(
+        Params._dummy(),
+        "optimizer_name",
+        "string representation of optimizer function for the underlying pytorch_lightning model",
+    )
+
+    pretrained = Param(
+        Params._dummy(),
+        "pretrained",
+        "True/False, whether to use pretrained weights for the backbone model or not",
+    )
+
+    feature_extracting = Param(
```

Comment:
might want to name this something active like extract_features, also im not sure what this does

Reply:
In LitDeepVisionModel: 
# Freeze those weights
if feature_extracting:
    for p in self.model.parameters():
        p.requires_grad = False

Should I also remove this param and always freeze all weights at the begining. And only free those layers users want to train explicitly?